sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Customizations/QuickPulseSDKClientAPIsRestClient.cs

```diff
@@ -0,0 +1,116 @@
+        /// <param name="xMsQpsTransmissionTime"> Timestamp when SDK transmits the metrics and documents to QuickPulse. A 8-byte long type of ticks. </param>
+        /// <param name="xMsQpsMachineName"> Computer name where AI SDK lives. QuickPulse uses machine name with instance name as a backup. </param>
+        /// <param name="xMsQpsInstanceName"> Service instance name where AI SDK lives. QuickPulse uses machine name with instance name as a backup. </param>
+        /// <param name="xMsQpsStreamId"> Identifies an AI SDK as trusted agent to report metrics and documents. </param>
+        /// <param name="xMsQpsRoleName"> Cloud role name for which SDK reports metrics and documents. </param>
+        /// <param name="xMsQpsInvariantVersion"> Version/generation of the data contract (MonitoringDataPoint) between SDK and QuickPulse. </param>
+        /// <param name="xMsQpsConfigurationEtag"> An encoded string that indicates whether the collection configuration is changed. </param>
+        /// <param name="monitoringDataPoint"> Data contract between SDK and QuickPulse. /QuickPulseService.svc/ping uses this as a backup source of machine name, instance name and invariant version. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="ikey"/> is null. </exception>
+        public ResponseWithHeaders<object, QuickPulseSDKClientAPIsPingHeaders> PingCustom(string ikey, string apikey = null, int? xMsQpsTransmissionTime = null, string xMsQpsMachineName = null, string xMsQpsInstanceName = null, string xMsQpsStreamId = null, string xMsQpsRoleName = null, string xMsQpsInvariantVersion = null, string xMsQpsConfigurationEtag = null, MonitoringDataPoint monitoringDataPoint = null, CancellationToken cancellationToken = default)
+        {
+            if (ikey == null)
+            {
+                throw new ArgumentNullException(nameof(ikey));
+            }
+
+            using var message = CreatePingRequest(ikey, apikey, xMsQpsTransmissionTime, xMsQpsMachineName, xMsQpsInstanceName, xMsQpsStreamId, xMsQpsRoleName, xMsQpsInvariantVersion, xMsQpsConfigurationEtag, monitoringDataPoint);
+            _pipeline.Send(message, cancellationToken);
+            var headers = new QuickPulseSDKClientAPIsPingHeaders(message.Response);
+            switch (message.Response.Status)
+            {
+                case 200:
+                    {
+                        CollectionConfigurationInfo value = default;
+                        if (message.Response.Headers.ContentLength != 0)
+                        {
+                            using var document = JsonDocument.Parse(message.Response.ContentStream);
+                            value = CollectionConfigurationInfo.DeserializeCollectionConfigurationInfo(document.RootElement);
+                        }
```

Comment:
Bug fix. Generated code doesn't check that ContentLength > 0 before deserializing.